tests/unit_tests/dao/datasource_test.py

```diff
@@ -0,0 +1,181 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+from sqlalchemy.orm.session import Session
+
+from superset.utils.core import DatasourceType
+
+
+def create_test_data(session: Session) -> None:
```

Comment:
You can also rewrite this as a fixture:

```python
@pytest.fixture
def session_with_data(session: Session) -> Iterator[Session]:
    ...  # code of `create_test_data` here
    yield session
```

And then in your tests:

```python
def test_get_datasource_sqlatable(app_context: None, session_with_data: Session) -> None:
    ...
```

But there's no need, a function like this also works fine.